### PR TITLE
docs: Add beginnings of troubleshooting doc.

### LINKED ISSUE #172: SPI flash erasers: document and make available
https://github.com/adafruit/Adafruit_SPIFlash/pull/1 has source and UF2 versions of an SPI flash erase program for all current Express boards. Accept this pull request if appropriate, and make the .uf2's available somewhere, along with documentation for why and how to use them.

Brief doc:

If your host computer starts complaining that your CIRCUITPY drive is corrupted, you can erase it completely. When CircuitPython restarts it will create a fresh empty CIRCUITPY filesystem.

**Warning: this will erase all your files in CIRCUITPY.**
1. Double-click the reset button.
2. Copy the appropriate .uf2 to the xxxBOOT drive.
3. The on-board NeoPixel  will turn blue, indicating the erase has started.
4. After about 15 seconds, the NexoPixel will start flashing green. If it flashes red, the erase failed.
5. Double-click again and load the appropriate CircuitPython .uf2

### PR DESCRIPTION
**Pull request by tannewt**
Fixes #172. Related to #214.

### CODE REVIEW

File: docs/troubleshooting.rst [hunk 1]
```diff
@@ -0,0 +1,40 @@
+Troubleshooting
+==============================
+
+From time to time, an error occurs when working with CircuitPython. Here are a
+variety of errors that can happen, what they mean and how to fix them.
+
+File system issues
+------------------------------
+
+If your host computer starts complaining that your CIRCUITPY drive is corrupted
```
Comment (willingc) [inline docs/troubleshooting.rst]: Let's put ``CIRCUITPY`` inside of ``

File: docs/troubleshooting.rst [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+or files cannot be overwritten or deleted, then you will have to erase it completely.
+When CircuitPython restarts it will create a fresh empty CIRCUITPY filesystem.
+
+This often happens on Windows when the CIRCUITPY disk is not safely ejected
+before being reset by the button or being disconnected from USB. This can also
+happen on Linux and Mac OSX but its less likely.
+
+.. danger:: This will erase all your files in CIRCUITPY!
```
Comment (willingc) [inline docs/troubleshooting.rst]: Perhaps:

```
.. caution:: Delete ``CIRCUITPY`` filesystem and reload CircuitPython.

    To reload CircuitPython (for example, to correct a corrupted filesystem),
    follow the process below. It's important to note that **any files stored on the
    ``CIRCUITPY`` drive will be erased**.
```

File: docs/troubleshooting.rst [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+#. Download the appropriate flash erase uf2 from `here <https://github.com/adafruit/Adafruit_SPIFlash/tree/master/examples/flash_erase_express>`_.
+#. Double-click the reset button.
+#. Copy the appropriate .uf2 to the xxxBOOT drive.
+#. The on-board NeoPixel will turn blue, indicating the erase has started.
+#. After about 15 seconds, the NexoPixel will start flashing green. If it flashes red, the erase failed.
+#. Double-click again and load the appropriate `CircuitPython .uf2 <https://github.com/adafruit/circuitpython/releases/latest>`_.
+
+ValueError: Incompatible .mpy file.
```
Comment (willingc) [inline docs/troubleshooting.rst]: `` around .mpy

File: docs/troubleshooting.rst [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+-------------------------------------
+
+This error occurs when importing a module that is stored as a `mpy` binary file
```
Comment (willingc) [inline docs/troubleshooting.rst]: `` around mypy and py

File: docs/troubleshooting.rst [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+(rather than a `py` text file) that was generated by a different version of
+CircuitPython than the one its being loaded into. Most versions are compatible
+but, rarely they aren't. In particular, the `mpy` binary format changed between
+CircuitPython versions 1.x and 2.x.
+
+So, if you just upgraded to 2.x from 1.x you'll need to download a newer version
```
Comment (willingc) [inline docs/troubleshooting.rst]: upgraded to CircuitPython 2.x..

File: docs/troubleshooting.rst [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+of the library. They are all available in the
```
Comment (willingc) [inline docs/troubleshooting.rst]: ~~library~~ Adafruit or Community bundle of libraries.

File: docs/troubleshooting.rst [hunk 7]
```diff
@@ -0,0 +1,40 @@
+Troubleshooting
+==============================
```
[the remark this answers is not in the document]
Comment (tannewt) [reply docs/troubleshooting.rst]: kk

File: docs/troubleshooting.rst (hunk 1 above)
Comment (tannewt) [reply docs/troubleshooting.rst]: Done.

File: docs/troubleshooting.rst (hunk 2 above)
Comment (tannewt) [reply docs/troubleshooting.rst]: Done.

File: docs/troubleshooting.rst (hunk 4 above)
Comment (tannewt) [reply docs/troubleshooting.rst]: done

File: docs/troubleshooting.rst (hunk 5 above)
Comment (tannewt) [reply docs/troubleshooting.rst]: done

File: docs/troubleshooting.rst (hunk 6 above)
Comment (tannewt) [reply docs/troubleshooting.rst]: I reworded this to make it clearer.

**Comment by tannewt** [comment]
Thanks for the suggestions!

**Comment by tannewt** [comment]
I'm going to pull this now so I can get the link setup. I'll be happy to fix any further comments in a follow up.

**Pull request merged**